templates/types/streaming/fastapi/app/engine/generate.py

```diff
@@ -54,14 +57,23 @@ def generate_datasource():
     ingestion_pipeline.vector_store = vector_store
 
     # Run the ingestion pipeline and store the results
-    ingestion_pipeline.run(show_progress=True, documents=documents)
+    nodes = ingestion_pipeline.run(show_progress=True, documents=documents)
 
     # Default vector store only keeps data in memory, so we need to persist it
     # Can remove if using a different vector store
     if isinstance(vector_store, SimpleVectorStore):
-        vector_store.persist(os.path.join(STORAGE_DIR, "vector_store.json"))
-    # Persist the docstore to apply ingestion strategy
-    docstore.persist(os.path.join(STORAGE_DIR, "docstore.json"))
+        index = VectorStoreIndex(
+            nodes=nodes,
+            storage_context=StorageContext.from_defaults(
+                docstore=docstore,
+                vector_store=vector_store,
+            ),
+            store_nodes_override=True,
+        )
+        index.storage_context.persist(STORAGE_DIR)
+    else:
+        # Persist the docstore to apply ingestion strategy
+        docstore.persist(os.path.join(STORAGE_DIR, "docstore.json"))
```

Comment:
any way to do this without specifying `docstore.json`? shouldn't it be the default? don't we need

Reply:
It's a path. Because we have custom storage config so we must construct the path instead of using default. The "docstore.json" is already default value, but let me make it better by importing the default variable instead.

Reply:
Using storage_context is fine, but it'll also generate other files like vector store, index store,.. which are redundant for other vector store. 
Explicitly persist for docstore is not a bad idea i think, it'll clearly show to the user that the ingestion pipeline for vector store only need the docstore.